core/server/themes/loader.js

```diff
@@ -0,0 +1,39 @@
+var debug = require('debug')('ghost:themes:loader'),
+    config = require('../config'),
+    events = require('../events'),
+    read = require('./read'),
+    settingsApi = require('../api/settings'),
+    updateConfigAndCache,
+    loadThemes,
+    initThemes;
+
+updateConfigAndCache = function updateConfigAndCache(themes) {
+    debug('loading themes', themes);
+    config.set('paths:availableThemes', themes);
+    settingsApi.updateSettingsCache();
+};
+
+loadThemes = function loadThemes() {
+    return read
+        .all(config.getContentPath('themes'))
+        .then(updateConfigAndCache);
+};
+
+initThemes = function initThemes() {
+    debug('init themes', settingsApi.cache.get('activeTheme'));
+
+    // Register a listener for server-start to load all themes
+    events.on('server:start', function readAllThemesOnServerStart() {
```

Comment:
One last question to this.
I assume this get's removed **soon** as well.
I assume we don't have to load the themes at all until they get requested.
The themes util could have a function which either loads the themes if not loaded yet or return if loaded?